event_registration_cancel_reason/wizard/event_registration_cancel_log_reason.py

```diff
@@ -21,9 +21,16 @@ class EventRegistrationCancelLogReason(models.TransientModel):
     def default_get(self, var_fields):
         res = super(EventRegistrationCancelLogReason, self).default_get(
             var_fields)
-        registration = self.env['event.registration'].browse(
-            self.env.context['active_id'])
-        res['event_type_id'] = registration.event_id.type.id
+        registrations = self.env['event.registration'].browse(
+            self.env.context['active_ids'])
+        event_type = registrations.mapped("event_id.type")
+        try:
+            event_type.ensure_one()
+        except exceptions.except_orm:
```

Comment:
Look, this will dissapear in the next version: https://github.com/odoo/odoo/blob/9.0/openerp/exceptions.py#L26, and we will need to change again the code. And I don't think the Python rule you're mentioning applies here, because this is not assume a field exists. This is checking if we have 1 record or more. Simply that.